app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkTouchHelper.kt

```diff
@@ -0,0 +1,111 @@
+import mozilla.components.support.ktx.android.content.getDrawableWithTint
+import mozilla.components.support.ktx.android.util.dpToPx
+import org.mozilla.fenix.R
+import org.mozilla.fenix.home.sessioncontrol.SwipeToDeleteCallback
+import org.mozilla.fenix.library.bookmarks.viewholders.BookmarkNodeViewHolder
+import org.mozilla.fenix.library.bookmarks.viewholders.BookmarkSeparatorViewHolder
+
+class BookmarkTouchHelper(interactor: BookmarkViewInteractor) :
+    ItemTouchHelper(BookmarkTouchCallback(interactor))
+
+class BookmarkTouchCallback(private val interactor: BookmarkViewInteractor) :
+    ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
+
+    override fun getSwipeDirs(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder
+    ): Int {
+        val item = (viewHolder as BookmarkNodeViewHolder).item
+        return if (viewHolder is BookmarkSeparatorViewHolder || item?.inRoots() == true) {
+            0
+        } else {
+            super.getSwipeDirs(recyclerView, viewHolder)
+        }
+    }
+
+    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
+        val item = (viewHolder as BookmarkNodeViewHolder).item
+        item?.let { interactor.onDelete(setOf(it)) }
+    }
+
+    override fun onChildDraw(
+        c: Canvas,
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        dX: Float,
+        dY: Float,
+        actionState: Int,
+        isCurrentlyActive: Boolean
+    ) {
+        super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive)
+        val icon = recyclerView.context.getDrawableWithTint(
+            R.drawable.ic_delete,
+            recyclerView.context.getColorFromAttr(R.attr.destructive)
+        )!!
+        val background = AppCompatResources.getDrawable(
+            recyclerView.context,
+            R.drawable.swipe_delete_background
+        )!!
+        val itemView = viewHolder.itemView
+        val iconLeft: Int
+        val iconRight: Int
+        val margin =
+            SwipeToDeleteCallback.MARGIN.dpToPx(recyclerView.context.resources.displayMetrics)
+        val iconWidth = icon.intrinsicWidth
+        val iconHeight = icon.intrinsicHeight
+        val cellHeight = itemView.bottom - itemView.top
+        val iconTop = itemView.top + (cellHeight - iconHeight) / 2
+        val iconBottom = iconTop + iconHeight
+
+        when {
+            dX > 0 -> { // Swiping to the right
+                iconLeft = itemView.left + margin
+                iconRight = itemView.left + margin + iconWidth
+                background.setBounds(
+                    itemView.left, itemView.top,
+                    (itemView.left + dX).toInt() + SwipeToDeleteCallback.BACKGROUND_CORNER_OFFSET,
+                    itemView.bottom
+                )
+                icon.setBounds(iconLeft, iconTop, iconRight, iconBottom)
```

Comment:
Since these two calls are made in every when scenario, let's pull them out to a little helper method

Reply:
"these two calls" = `background.setBounds()` and `icon.setBounds()`